Tasks/NuGetCommandV2/nugetpublisher.ts

```diff
@@ -98,18 +98,22 @@ export async function run(nuGetPath: string): Promise<void> {
         // Setting up auth info
         let accessToken = auth.getSystemAccessToken();
         const quirks = await ngToolRunner.getNuGetQuirksAsync(nuGetPath);
-        let credProviderPath = nutil.locateCredentialProvider();
+
         // Clauses ordered in this way to avoid short-circuit evaluation, so the debug info printed by the functions
         // is unconditionally displayed
-        let useCredProvider = ngToolRunner.isCredentialProviderEnabled(quirks) && credProviderPath;
-        let useCredConfig = ngToolRunner.isCredentialConfigEnabled(quirks) && !useCredProvider;
+        const useV1CredProvider: boolean = ngToolRunner.isCredentialProviderEnabled(quirks);
+        const useV2CredProvider: boolean = ngToolRunner.isCredentialProviderV2Enabled(quirks);
+        const credProviderPath: string = nutil.locateCredentialProvider(useV2CredProvider);
+        const useCredConfig = ngToolRunner.isCredentialConfigEnabled(quirks) && (!useV1CredProvider && !useV2CredProvider);
 
-        const internalAuthInfo = new auth.InternalAuthInfo(urlPrefixes, accessToken, useCredProvider, useCredConfig);
+        const internalAuthInfo = new auth.InternalAuthInfo(urlPrefixes, accessToken, ((!useV1CredProvider && !useV2CredProvider) ? credProviderPath : null), useCredConfig);
```

Comment:
This condition doesn't look right. In case one provider is enabled, it will set the path to null instead of the correct one. We should run the NuGet L2s with these changes to make sure everything is still passing.